gitnamehere/point-of-sale-group-project
# Employee Account Creation & Modification

Bosses can now create employee accounts

Ticket: [ATS-56](https://cs375-pos-group-project.atlassian.net/jira/core/projects/ATS/board?selectedIssue=ATS-56)

- Created input fields for employee information:
   - First Name
   - Last Name
   - Username
   - Password
- Refactored API endpoint for creating employee account `api/auth/pos/account/create`
- Fixed bug where customer account endpoint required authentication

Testing:
Account Creation:
- npm run start:dev
- log into pos
- go to Settings page
- Verify that "Create Account" button is in navbar
- Click "Create Account" button
- Verify that it routes to `http://localhost:3000/pos/account-management/create.html`
- Verify that "Modify Account" button is in navbar on Create Account page
- Create valid account
- Verify that "Success" appears at top of page with 200 status code in Network tab
- Go into psql database
- Verify that account has been added to the `account` table with all relevant information
- Verify that account password has been hashed
- Log out of pos
- Try to log into pos with newly created account
- Verify that employee account can log in

Account Modification:
- log into pos
- go to Settings page
- Verify that "Modify Account" button is in navbar
- Click "Modify Account" button
- Verify that it routes to `http://localhost:3000/pos/account-management/update.html`
- Verify that all accounts in database are present in dropdown
- Verify that selecting an account from dropdown auto-populates data in fields
- Make edits to an account
- - Change an account username to a new username that **doesn't** already exist in the database
- Enter **incorrect password** for that account and click the "Update Account" button
- Verify that "Bad Request" appears at top of page with 400 status code in Network tab
- Enter **correct password** for that account and click the "Update Account" button
- Verify that "Success" appears at top of page with 200 status code in Network tab
- Go into psql database
- Verify that account has been modified in the `account` table with all new information
- Go back to `http://localhost:3000/pos/account-management/update.html`
- Make edits to an account
- - Change an account username to a new username that **does** already exist in the database
- Enter password for that account and click the "Update Account" button
- Verify that "Username Already Exists" appears at top of page with 400 status code in Network tab
